octopusdeploy_framework/datasource_space.go

```diff
@@ -14,39 +15,28 @@ type spaceDataSource struct {
 	*Config
 }
 
-type spaceModel struct {
-	ID                       types.String `tfsdk:"id"`
-	Name                     types.String `tfsdk:"name"`
-	Slug                     types.String `tfsdk:"slug"`
-	Description              types.String `tfsdk:"description"`
-	IsDefault                types.Bool   `tfsdk:"is_default"`
-	SpaceManagersTeams       types.List   `tfsdk:"space_managers_teams"`
-	SpaceManagersTeamMembers types.List   `tfsdk:"space_managers_team_members"`
-	IsTaskQueueStopped       types.Bool   `tfsdk:"is_task_queue_stopped"`
-}
-
 func NewSpaceDataSource() datasource.DataSource {
 	return &spaceDataSource{}
 }
 
-func (*spaceDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
-	resp.TypeName = ProviderTypeName + "_space"
+func (*spaceDataSource) Metadata(_ context.Context, _ datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = util.GetTypeName("space")
 }
 
-func (*spaceDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+func (*spaceDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
 	resp.Schema = schema.Schema{
 		Description: "Provides information about an existing space.",
-		Attributes:  getSpaceSchema(),
+		Attributes:  schemas.GetSpaceDatasourceSchema(),
 	}
 }
 
-func (b *spaceDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+func (b *spaceDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
 	b.Config = DataSourceConfiguration(req, resp)
 }
 
 func (b *spaceDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
 	var err error
-	var data spaceModel
+	var data schemas.SpaceModel
 	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
```

Comment:
I'm still confused on these, I thought for reads we would be using State. [The docs seem to use State.](https://developer.hashicorp.com/terraform/plugin/framework/resources/read)